crates/red_knot_workspace/src/db/changes.rs

```diff
@@ -130,6 +130,17 @@ impl RootDatabase {
                     }
                 }
 
+                watch::ChangeEvent::CreatedVirtual(path)
+                | watch::ChangeEvent::ChangedVirtual(path) => {
+                    File::sync_virtual_path(self, &path);
+                }
+
+                watch::ChangeEvent::DeletedVirtual(path) => {
+                    if let Some(virtual_file) = self.files().try_virtual_file(&path) {
+                        virtual_file.close(self);
+                    }
+                }
```

Comment:
Or, should we expose a `close_virtual_file` on `Files` instead?

Reply:
I think this is fine for now